karapace/backup/consumer.py

```diff
@@ -0,0 +1,45 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from datetime import timedelta
+from isodate import duration_isoformat, parse_duration
+from typing import Union
+
+__all__ = ["PollTimeout"]
+
+
+# TODO @final (Python 3.8+)
```

Comment:
question: Is it important somehow that this class isn't subclassed? Unless something breaks when subclassing, I'd consider not adding `@final` at all. (And if it is important, maybe add a comment explaining why?)

Reply:
By not making it final we have to assume that someone using Karapace is subclassing this class, and using it. This has serious implications when it comes to making changes to public symbols. Everything should be final, to avoid such issues.

Reply:
I'm failing to see any serious implications at all. 

Karapace is an application, and not a library, so any usage in that way is obviously not supported and would have no compatibility guarantees whatsoever in the first place. Internals like this is not a public API.

Basically I disagree that we should assume this application is used as a library.